src/client/scripts/game/gui/guipractice.js

```diff
@@ -0,0 +1,213 @@
+        style.hideElement(element_menuExternalLinks);
+        closeListeners()
+    }
+
+    function initListeners() {
+        element_practiceBack.addEventListener('click', callback_practiceBack)
+        element_checkmatePractice.addEventListener('click', callback_checkmatePractice)
+        element_tacticsPractice.addEventListener('click', gui.callback_featurePlanned)
+        element_practicePlay.addEventListener('click', callback_practicePlay)
+        for (let element of elements_checkmates) {
+            element.addEventListener('click', callback_checkmateList);
+        }
+    }
+
+    function closeListeners() {
+        element_practiceBack.removeEventListener('click', callback_practiceBack)
+        element_checkmatePractice.removeEventListener('click', callback_checkmatePractice)
+        element_tacticsPractice.removeEventListener('click', gui.callback_featurePlanned)
+        element_practicePlay.removeEventListener('click', callback_practicePlay)
+        for (let element of elements_checkmates) {
+            element.removeEventListener('click', callback_checkmateList);
+        }
+    }
+
+    function changePracticeMode(mode) { // checkmate-practice / tactics-practice
+        modeSelected = mode
+        if (mode === 'checkmate-practice') {
+            element_practiceName.textContent = translations["menu_checkmate"]
+            element_checkmatePractice.classList.add('selected')
+            element_tacticsPractice.classList.remove('selected')
+            element_checkmatePractice.classList.remove('not-selected')
+            element_tacticsPractice.classList.add('not-selected')
+            // callback_updateOptions()
+        } else if (mode === 'tactics-practice') {
+            // nothing yet
+        }
+    }
+
+    function changeCheckmateSelected(checkmateid) {
+        for (let element of elements_checkmates){
+            if (checkmateid === element.id) {
+                element.classList.remove('not-selected')
+                element.classList.add('selected')
+                checkmateSelectedID = checkmateid;
+            } else {
+                element.classList.remove('selected')
+                element.classList.add('not-selected')
+            }
+        }
+    }
+
+    // TODO: implement beaten checkmate list
+    function updateCheckmatesBeaten() {
+        for (let element of elements_checkmates){
+            element.classList.remove('beaten')
+            element.classList.add('unbeaten')
+        }
+
+        // elements_checkmates[2].classList.add('beaten')
+        // elements_checkmates[2].classList.remove('unbeaten')
+    }
+
+    function callback_practiceBack(event) {
+        event = event || window.event;
+        close()
+        guititle.open()
+    }
+
+    function callback_checkmatePractice(event) {
+        event = event || window.event;
+        changePracticeMode('checkmate-practice')
+    }
+
+    function callback_checkmateList(event){
+        event = event || window.event;
```

Comment:
The lines "event = event || window.event;" are depricated, we can delete them all. I've been needing to do that.